selene_sdk/samplers/multi_file_sampler.py

```diff
@@ -63,17 +74,34 @@ def __init__(self,
             output_dir=output_dir)
 
         self._samplers = {
-            "train": train_sampler,
-            "validate": validate_sampler
+            "train": train_sampler if isinstance(train_sampler, Sampler) \
+                     else None,
+            "validate": validate_sampler if isinstance(validate_sampler, Sampler) \
+                        else None
+        }
+
+        self._dataloaders = {
+            "train": train_sampler if isinstance(train_sampler, DataLoader) \
+                     else None,
+            "validate": validate_sampler if isinstance(validate_sampler, DataLoader) \
+                        else None
         }
 
-        self._index_to_feature = {
-            i: f for (i, f) in enumerate(features)
+        self._iterators = {
+            "train": iter(self._dataloaders["train"]) \
+                if self._dataloaders["train"] else None,
+            "validate": iter(self._dataloaders["validate"]) \
+                if self._dataloaders["validate"] else None
         }
 
+        self._index_to_feature = {i: f for (i, f) in enumerate(features)}
```

Comment:
should this be targets ?

Reply:
The code was actually still using features instead of targets